File: Leetcode 2206 Divide Array Into Equal Pairs.go
// 2026/08/20

func divideArray(nums []int) bool {
    n_map := make(map[int]int)
    for _, num := range nums{
        n_map[num] ++
    }
    // fmt.Println(n_map)
    for _, c := range n_map {
        if c % 2 == 1 {
            return false
        }
    }

    return true
}

File: Leetcode 2206 Divide Array Into Equal Pairs.py
# 2026/08/20
class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        n_map = {}
        for i in nums:
            if i in n_map:
                n_map[i] += 1
            else:
                n_map[i] = 1
        for index, item in n_map.items():
            if item % 2 == 1:
                return False
        return True
